src/promptflow-devkit/promptflow/_sdk/_utilities/chat_utils.py

```diff
@@ -10,12 +19,127 @@ def construct_flow_absolute_path(flow: str) -> str:
 
 
 # Todo: use base64 encode for now, will consider whether need use encryption or use db to store flow path info
-def construct_chat_page_url(flow, port, url_params, enable_internal_features=False):
-    flow_path_dir, flow_path_file = resolve_flow_path(flow)
-    flow_path = str(flow_path_dir / flow_path_file)
+def construct_chat_page_url(flow_path, port, url_params, enable_internal_features=False):
     encrypted_flow_path = encrypt_flow_path(flow_path)
     query_dict = {"flow": encrypted_flow_path}
     if enable_internal_features:
         query_dict.update({"enable_internal_features": "true", **url_params})
     query_params = urlencode(query_dict)
     return urlunparse(("http", f"127.0.0.1:{port}", "/v1.0/ui/chat", "", query_params, ""))
+
+
+content_cache = defaultdict(dict)
+file_meta_cache = defaultdict(dict)
+
+
+def any_file_update(monitor_target, ignore=None):
+    global file_meta_cache
+    updated = False
+
+    for sub in monitor_target.rglob("**/*"):
```

Comment:
flow test may generate some files under .promptflow/xx, might need to skip those files

Reply:
they have been skipped by ignore

Reply:
but seems that there is no easy way to avoid iterate them